### PR TITLE
Aruha 237 uncompress request body

### PR DESCRIPTION
**Pull request by v-stepanov**
@tmuehl @rcillo @antban @jkliff Please review

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/zalando/nakadi/pull/293" target="_blank">ReviewNinja

**Comment by v-stepanov** [comment]
(ninja star is a lie)

File: src/main/java/de/zalando/aruha/nakadi/util/GzipBodyRequestFilter.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        @Override
+        public void close() throws IOException {
+            inputStream.close();
+        }
+
+        @Override
+        public boolean isFinished() {
+            try {
+                return inputStream.available() == 0;
+            } catch (IOException e) {
```
Comment (antban) [inline src/main/java/de/zalando/aruha/nakadi/util/GzipBodyRequestFilter.java]: final

File: src/main/java/de/zalando/aruha/nakadi/util/GzipBodyRequestFilter.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    private static final Logger LOG = LoggerFactory.getLogger(GzipBodyRequestFilter.class);
+
+    @Override
+    public final void doFilter(final ServletRequest servletRequest, final ServletResponse servletResponse,
+                               final FilterChain chain) throws IOException, ServletException {
+
+        HttpServletRequest request = (HttpServletRequest) servletRequest;
+        final HttpServletResponse response = (HttpServletResponse) servletResponse;
+
+        boolean isGzipped = Optional
```
Comment (antban) [inline src/main/java/de/zalando/aruha/nakadi/util/GzipBodyRequestFilter.java]: final

File: src/main/java/de/zalando/aruha/nakadi/util/GzipBodyRequestFilter.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+public class GzipBodyRequestFilter implements Filter {
+
+    private static final Logger LOG = LoggerFactory.getLogger(GzipBodyRequestFilter.class);
+
+    @Override
+    public final void doFilter(final ServletRequest servletRequest, final ServletResponse servletResponse,
+                               final FilterChain chain) throws IOException, ServletException {
+
+        HttpServletRequest request = (HttpServletRequest) servletRequest;
+        final HttpServletResponse response = (HttpServletResponse) servletResponse;
```
Comment (antban) [inline src/main/java/de/zalando/aruha/nakadi/util/GzipBodyRequestFilter.java]: Why do you need to cast? (Or why do you have this variable at all?)

File: src/main/java/de/zalando/aruha/nakadi/util/GzipBodyRequestFilter.java (hunk 1 above)
Comment (v-stepanov) [inline src/main/java/de/zalando/aruha/nakadi/util/GzipBodyRequestFilter.java]: fixed

File: src/main/java/de/zalando/aruha/nakadi/util/GzipBodyRequestFilter.java (hunk 2 above)
Comment (v-stepanov) [inline src/main/java/de/zalando/aruha/nakadi/util/GzipBodyRequestFilter.java]: fixed

File: src/main/java/de/zalando/aruha/nakadi/util/GzipBodyRequestFilter.java (hunk 3 above)
Comment (v-stepanov) [inline src/main/java/de/zalando/aruha/nakadi/util/GzipBodyRequestFilter.java]: no real need for that variable. Removed it

File: src/main/java/de/zalando/aruha/nakadi/config/WebConfig.java [hunk 4]
```diff
@@ -45,18 +47,17 @@ public TimeoutCallableProcessingInterceptor timeoutInterceptor() {
 
     @Bean
     public FilterRegistrationBean flowIdRequestFilter() {
-        final FilterRegistrationBean filterRegistrationBean = new FilterRegistrationBean();
-        filterRegistrationBean.setFilter(new FlowIdRequestFilter());
-        filterRegistrationBean.setOrder(Ordered.HIGHEST_PRECEDENCE + 1);
-        return filterRegistrationBean;
+        return createFilterRegistrationBean(new FlowIdRequestFilter(), Ordered.HIGHEST_PRECEDENCE + 2);
```
Comment (tmuehl) [inline src/main/java/de/zalando/aruha/nakadi/config/WebConfig.java]: Me, personally, I think I would execute the FlowIdRequestFilter first and the GzipBodyRequestFilter afterwards. If something during the decompression goes wrong, you already have the FlowId in the logs.

Comment (v-stepanov) [inline src/main/java/de/zalando/aruha/nakadi/config/WebConfig.java]: Makes sense 👍 
Fixed

File: src/acceptance-test/java/de/zalando/aruha/nakadi/webservice/CompressedEventPublishingAT.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+import java.util.zip.GZIPOutputStream;
+
+import static com.jayway.restassured.RestAssured.given;
+import static com.jayway.restassured.http.ContentType.JSON;
+import static de.zalando.aruha.nakadi.utils.TestUtils.buildEventType;
+import static de.zalando.aruha.nakadi.utils.TestUtils.randomValidEventTypeName;
+import static java.text.MessageFormat.format;
+import static javax.ws.rs.core.HttpHeaders.CONTENT_ENCODING;
+
+public class CompressedEventPublishingAT extends BaseAT {
```
Comment (rcillo) [inline src/acceptance-test/java/de/zalando/aruha/nakadi/webservice/CompressedEventPublishingAT.java]: Maybe you could also exercise the path that generates a error response (compression with GET, for example).

Comment (v-stepanov) [inline src/acceptance-test/java/de/zalando/aruha/nakadi/webservice/CompressedEventPublishingAT.java]: Very good point, as it was actually returning 500 in that case :-D (which is not what we want)
I now fixed that and added AT for that.

**Comment by rcillo** [comment]
:shipit:

**Pull request merged**